flang/include/flang/Lower/Support/BoxValue.h

```diff
@@ -95,9 +95,11 @@ class AbstractArrayBox {
 public:
   AbstractArrayBox() = default;
   AbstractArrayBox(llvm::ArrayRef<mlir::Value> extents,
-                   llvm::ArrayRef<mlir::Value> lbounds)
+                   llvm::ArrayRef<mlir::Value> lbounds, mlir::Value slice,
+                   bool sliceIsOneBased)
       : extents{extents.begin(), extents.end()}, lbounds{lbounds.begin(),
-                                                         lbounds.end()} {}
+                                                         lbounds.end()},
+        slice{slice}, sliceIsOneBased{sliceIsOneBased} {}
```

Comment:
What does `sliceIsOneBased` mean? (I'm commenting as I go, so this may be cleared up down below...)